src/node/dbus.c

```diff
@@ -0,0 +1,44 @@
+#include "dbus.h"
+
+#include <arpa/inet.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+int setup_peer_dbus(sd_bus *dbus, const struct sockaddr_in *addr) {
+        int r;
+        r = sd_bus_new(&dbus);
+        if (r < 0) {
+                fprintf(stderr, "Failed to create bus: %s\n", strerror(-r));
+                return -1;
+        }
+        (void) sd_bus_set_description(dbus, "orchestrator");
+
+        /* we trust everything from the orchestrator, there is only one peer anyway */
+        r = sd_bus_set_trusted(dbus, true);
+        if (r < 0) {
+                fprintf(stderr, "Failed to trust orchestrator: %s\n", strerror(-r));
+                return -1;
+        }
+
+        _cleanup_free_ char *dbus_addr = NULL;
+        r = asprintf(&dbus_addr, "tcp:host=%s,port=%d", inet_ntoa(addr->sin_addr), ntohs(addr->sin_port));
+        if (r < 0) {
+                fprintf(stderr, "Out of memory\n");
+                return -1;
+        }
+
+        r = sd_bus_set_address(dbus, dbus_addr);
+        if (r < 0) {
+                fprintf(stderr, "Failed to set address: %s\n", strerror(-r));
+                return -1;
+        }
+
+        fprintf(stdout, "connecting to orchestrator on '%s'\n", dbus_addr);
+        r = sd_bus_start(dbus);
+        if (r < 0) {
+                fprintf(stderr, "Failed to connect to orchestrator: %s\n", strerror(-r));
+                return -1;
+        }
+
+        return 0;
```

Comment:
Stupid question: 

In some parts of the code we `return 0` (or another appropriate number) and in different parts of code we `return EXIT_SUCCESS` (or failure). Is there some functional difference between those? Google doesn't help very much

Reply:
Is this something that would make sense to add to the code style?

Reply:
Yeah, this is how it's done.
And regarding strcmp - it actually makes quite a bit of sense there. In Java for example, String::compareTo() does the same